### PR TITLE
Remember entities starting separation of concerns

### PR DESCRIPTION
**Pull request by johanandren**
Move all logic for starting up remembered entities (at once or slow) out of the `Shard` into the `RememberEntitiesStarter` and eliminate the future-impl by using actor timer/messaging/state.

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala [hunk 1]
```diff
@@ -14,8 +14,11 @@ import akka.annotation.InternalApi
 import akka.cluster.sharding.ClusterShardingSettings
 import akka.cluster.sharding.Shard
 import akka.cluster.sharding.ShardRegion
+import akka.cluster.sharding.ShardRegion.EntityId
```
Comment (chbatey) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala:17]: match up file name with the class name?

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala [hunk 2]
```diff
@@ -30,7 +33,8 @@ private[akka] object RememberEntityStarter {
       settings: ClusterShardingSettings) =
     Props(new RememberEntityStarter(region, shard, shardId, ids, settings))
 
-  private case object Tick extends NoSerializationVerificationNeeded
+  private final case class StartBatch(batchSize: Int) extends NoSerializationVerificationNeeded
```
Comment (chbatey) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala:36]: if the batch size is always the same can this be a case object

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala [hunk 3]
```diff
@@ -47,35 +51,75 @@ private[akka] final class RememberEntityStarter(
...
+      if (entitiesMoved.nonEmpty) {
+        log.info("Found [{}] entities moved to new shard(s)", entitiesMoved.size)
+        shard ! Shard.EntitiesMovedToOtherShard(entitiesMoved)
       }
+      context.stop(self)
+    }
+  }
+
+  private def startBatch(batchSize: Int): Unit = {
+    log.debug("Starting batch of [{}] remembered entities", batchSize)
```
Comment (chbatey) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala:101]: do we want a max outstanding requests?

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala (hunk 2 above)
Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala:36]: It's from user config

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala (hunk 3 above)
Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala:101]: Could make sense but I think we can improve that later if we want it, this is pretty much exactly how it works today.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala (hunk 1 above)
Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/RememberEntitiesStarter.scala:17]: Ah, ups, will do!

**Review by chbatey — APPROVED** [review]

**Pull request merged**